src/hotspot/cpu/riscv/c1_LIRAssembler_riscv.cpp

```diff
@@ -1637,13 +1637,10 @@ void LIR_Assembler::emit_updatecrc32(LIR_OpUpdateCRC32* op) {
   assert_different_registers(val, crc, res);
   __ la(res, ExternalAddress(StubRoutines::crc_table_addr()));
 
-  __ xori(crc, crc, -1); // ~crc
-  __ slli(crc, crc, 32);
-  __ srli(crc, crc, 32);
+  __ notr(crc, crc); // ~crc
+  __ zero_extend(crc, crc, 32);
   __ update_byte_crc32(crc, val, res);
-  __ xori(res, crc, -1); // ~crc
-  __ slli(crc, crc, 32);
-  __ srli(crc, crc, 32);
+  __ notr(res, crc); // ~crc
```

Comment:
Do you miss the `zero_extend(crc, crc, 32)`?

Reply:
As far as I can see this is unneeded, actually. I used the `test/hotspot/jtreg/compiler/codegen/CRCTest.java` test as a sanity check (it uses the `emit_updatecrc32` stub) and it's ok.

Reply:
Seems not, it needs 32 bits only.

Reply:
> Do you miss the zero_extend(crc, crc, 32)?

seems not, it needs 32 bits only.